thrift/idl.go

```diff
@@ -371,104 +373,138 @@ func addFunction(ctx context.Context, fn *parser.Function, tree *parser.Thrift,
 
 	}
 
+	st, err := streaming.ParseStreaming(fn)
+	if err != nil {
+		return err
+	}
+	isStreaming := st.ClientStreaming || st.ServerStreaming
```

Comment:
what if a  method used both on `ClientStreaming` and `ServerUnary`? what's the behavior?

Reply:
> method used both on ClientStreaming and ServerUnary

Do you mean what will happen when the streaming.mode is "client" in the thrift idl?
like `Response EchoClient (1: Request req) (streaming.mode="client")`

Reply:
wait, I got what your question meant. Let me sort out

Reply:
When the method is a streaming method (either of client streaming, server streaming, bidirectional streaming), both request and response descriptor should be unwrapped.
It doesn't matter with the mode of req and resp. It depends on if the method (function) is streaming or unary.